### PR TITLE
Update chain ID fns and add from method for 8-byte typed chain bytes

### PR DESCRIPTION
**Pull request by drewstone**

### CODE REVIEW

File: proposals/src/header.rs [hunk 1]
```diff
@@ -79,7 +79,15 @@ impl TypedChainId {
...
     /// Get the chain id as a `u32`.
     #[must_use]
-    pub const fn chain_id(&self) -> u32 {
+    pub fn chain_id(&self) -> u64 {
+        let mut buf: [u8; 8] = [0u8; 8];
+        buf[2..8].copy_from_slice(&self.to_bytes());
+        u64::from_be_bytes(buf)
+    }
+
+    /// Get the chain id as a `u32`.
```
Comment (dharjeezy) [inline proposals/src/header.rs:88]: the comment should be u64

Comment (drewstone) [reply proposals/src/header.rs]: Updated thanks!

File: proposals/src/header.rs [hunk 2]
```diff
@@ -163,6 +175,24 @@ impl From<[u8; Self::LENGTH]> for TypedChainId {
...
+            0x0200 => TypedChainId::Substrate(id),
+            0x0301 => TypedChainId::PolkadotParachain(id),
+            0x0302 => TypedChainId::KusamaParachain(id),
+            0x0303 => TypedChainId::RococoParachain(id),
+            0x0400 => TypedChainId::Cosmos(id),
+            0x0500 => TypedChainId::Solana(id),
+            _ => Self::None,
+        }
+    }
+}
```
Comment (shekohex) [inline proposals/src/header.rs:194]: Could we also have the `impl From<u64> for TypedChainId`? it should be fairly easy as converting the `u64` to bytes and calling the `TypedChainid::from(u64_bytes)`

**Review by shekohex — CHANGES_REQUESTED** [review]
Looks good! just adding a few more functions and good to go.

File: proposals/src/header.rs (hunk 2 above)
Comment (drewstone) [reply proposals/src/header.rs:194]: Added!

**Review by shekohex — APPROVED** [review]
LGTM!

**Pull request merged**